scripts/js/lib/links/extractAltText.ts

```diff
@@ -0,0 +1,49 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { visit } from "unist-util-visit";
+import { unified } from "unified";
+import { Root } from "remark-mdx";
+import remarkStringify from "remark-stringify";
+import rehypeRemark from "rehype-remark";
+import rehypeParse from "rehype-parse";
+import remarkGfm from "remark-gfm";
+ 
+export interface AltText {
+  imageName: Set<string>; 
+  altText: Set<string | null | undefined>;
+};
+
+
+export async function extractAlt(
+  markdown: string,
+): Promise<AltText> {
```

Comment:
You can simply this functionality by making it more focused. All we care about is finding which images are missing alt text. We don't need to keep track of which images do have alt text. Note that if you are missing alt text, then `altText` would be empty. So, all we need is to return a `string[]` for the image names we're missing. You could rename the function something like `findImagesWithoutAltText`